### PR TITLE
ON-29697 # Added element replacement regex logic

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: src/formElementsService.ts [hunk 1]
```diff
@@ -212,10 +212,49 @@ function flattenFormElements(
   )
 }
 
+/**
+ * The regex used for matching `{ELEMENT:<elementName>}` tags in OneBlink
+ * Calculation and Info (HTML) elements.
+ */
+const WYSIWYGRegex = /({ELEMENT:)([^}]+)(})/g
```
Comment (divporter) [inline src/formElementsService.ts:219]: also you'll need to do string.MatchAll to use capture groups with a global flag
```suggestion
const WYSIWYGRegex = /({ELEMENT:)(.*)(})/g
```

File: src/submissionService.ts [hunk 2]
```diff
@@ -41,3 +41,47 @@ export function getRootElementValueById(
...
+ *
+ * @param elementName
+ * @param submission
+ * @returns
+ */
+export function getSubmissionValueAsString(
+  elementName: string,
+  submission: SubmissionTypes.S3SubmissionData['submission'],
+): string {
+  const v = submission[elementName]
```
Comment (divporter) [inline src/submissionService.ts:67]: During a submission the value could be on the model but not shown, don't know if it's here we want to take care of that on in apps-react.  It's not an issue post submission as the model will have been scrubbed.

Comment (Zaxist) [reply src/submissionService.ts:67]: Mm,yeh I think we handle it in apps-react, and pass in the clean submission model

Comment (divporter) [reply src/submissionService.ts:67]: I would just use the parent form elements conditionally shown object, much faster.

**Pull request merged**